### ISSUE TITLE
IntelliSense support for MSTest.Sdk properties

### ISSUE THREAD

**Issue by Youssef1313** [microsoft/testfx#4118]
Currently, there doesn't seem to be any IntelliSense support in csproj. I think it's useful to have such support.

I'm seeing all MSBuild IntelliSense defined [here](https://github.com/dotnet/msbuild/blob/e73ffcba1fa42ca60551be5dadbcf05c9ad9d914/src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd) (even for [Razor](https://github.com/dotnet/msbuild/blob/e73ffcba1fa42ca60551be5dadbcf05c9ad9d914/src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd#L2220)).

So, we could just PR to MSBuild to add IntelliSense for MSTest.Sdk.

However, it will be a better idea if we can maintain the xsd ourselves and have it shipped inside MSTest.Sdk NuGet package. This will allow us to always have the right IntelliSense depending on the actual MSTest.Sdk version that the user is using! But I don't see a way MSBuild would allow it, which is probably Razor is on MSBuild side as well.

List of properties to have IntelliSense for:

- `UseVSTest` (bool)
- `EnableMSTestRunner` (bool)
- `GenerateTestingPlatformEntryPoint` (bool)
- `GenerateSelfRegisteredExtensions` (bool)
- `GenerateProgramFile` (bool)
- `DisableTestingPlatformServerCapability` (bool)
- `TestingExtensionsProfile` (`Default`, `None`, or `AllMicrosoft`)
- `EnableMicrosoftTestingExtensionsCrashDump` (bool)
- `EnableMicrosoftTestingExtensionsCodeCoverage` (bool)
- `EnableMicrosoftTestingExtensionsHangDump` (bool)
- `EnableMicrosoftTestingExtensionsHotReload` (bool)
- `EnableMicrosoftTestingExtensionsRetry` (bool)
- `EnableMicrosoftTestingExtensionsTrxReport` (bool)
- `EnableAspireTesting` (bool)
- `EnablePlaywright` (bool)
- `IsTestingPlatformApplication` (bool)
- `TestingPlatformCaptureOutput` (bool)
- `TestingPlatformCommandLineArguments` (string - arbitrary user value)
- `TestingPlatformDotnetTestSupport` (bool)
- `TestingPlatformShowTestsFailure` (bool)

**Comment by Youssef1313** [comment]
@rainersigwald Can you advise please if there is a way to provide our own xsd? Otherwise, are you okay with updating CommonTypes.xsd in MSBuild repo?

**Comment by Evangelink** [comment]
I would also consider `EnableMSTestRunner` (bool) and all these properties https://github.com/microsoft/testfx/blob/main/src/Adapter/MSTest.TestAdapter/build/net/MSTest.TestAdapter.targets#L5-L8

@Youssef1313 I'll let you edit the lsit above

**Comment by rainersigwald** [comment]
Unfortunately there's not currently a way to provide your own XSD--but we'd be happy to accept a PR to add these to CommonTypes.xsd.

**Comment by Youssef1313** [comment]
@Evangelink Should we consider also the TestingPlatformBuilderHook item?

**Comment by Evangelink** [comment]
> @Evangelink Should we consider also the TestingPlatformBuilderHook item?

I don't think so. It's quite technical and will be used by really few people (extension authors). We have documented it in the tech doc of the platform.

**Comment by Youssef1313** [comment]
Closing out as msbuild PR is now merged.
